### DISCUSSION TITLE
What is the better `render` test function?

### DISCUSSION THREAD

**Discussion by The-Lum** [plantuml/plantuml#1611] [Q&A]
Hi all,

- What is the better `render` test function?
_Or perhaps it will be goood to factorize some `render` functions in order to normalize and help plantuml test..._
_[in order to specialize test: pre-processing, image, layout, output, option, ...]_

## Link
- [PlantUML API](https://plantuml.com/api)

**Comment by arnaudroques** [comment]
> * What is the better `render` test function?

Hello! Not sure to know to which render test function you are referring to...

About testing, a possible way is to use way that we use in A0000_Test/A0000_TestResult, that is a special Ugraphic which "render" the diagram in some text format. Tell us if you need more explanation about this.

About the API, the current one is really ugly :-)
It is something that definitively need to be improved, so any new idea is welcome!

**Reply by The-Lum** [reply]
Ok, I will search `render` or `runPlatnUML` on test dir. :-)

**Reply by The-Lum** [reply]
👀
Well: on `Test_Simple` and `BasicTest.java`, you use `FileFormat.DEBUG`.
Perhaps we can factorize that and manage more FileFormat... _[in order to specialize test: pre-processing, image, layout, output, option, ...]_
